src/extensions/score_source_code_linker/__init__.py

```diff
@@ -59,6 +59,26 @@ def setup(app: Sphinx) -> dict[str, str | bool]:
     }
 
 
+
+def find_need(
+    all_needs: NeedsMutable, id: str, prefixes: list[str]
+) -> NeedsInfoType | None:
+    """
+    Checks all possible external 'prefixes' for an ID
+    So that the linker can add the link to the correct NeedsInfoType object.
+    """
+    if id in all_needs:
+        return all_needs[id]
+
+    # Try all possible prefixes
+    for prefix in prefixes:
+        prefixed_id = f"{prefix}{id}"
+        if prefixed_id in all_needs:
+            return all_needs[prefixed_id]
+
+    return None
+
+
 # re-qid: gd_req__req__attr_impl
 def add_source_link(app: Sphinx, env: BuildEnvironment) -> None:
```

Comment:
This is executed only once?!

```suggestion
def add_source_links(app: Sphinx, env: BuildEnvironment) -> None:
```